integration-cli/docker_cli_build_test.go

```diff
@@ -6740,3 +6740,30 @@ func (s *DockerSuite) TestBuildLabelsOverride(c *check.C) {
 	}
 
 }
+
+// #20083
+func (s *DockerSuite) TestBuildDockerignoreComment(c *check.C) {
+	name := "testbuilddockerignorecleanpaths"
+	dockerfile := `
+        FROM busybox
+        ADD . /tmp/
+        RUN sh -c "(! ls /tmp/foo) && (! ls /tmp/foo2) && (ls /tmp/dir1/foo)"`
+	ctx, err := fakeContext(dockerfile, map[string]string{
+		"foo":      "foo",
+		"foo2":     "foo2",
+		"dir1/foo": "foo in dir1",
+		".dockerignore": `# Visual C++ cache files
+# because we have git ;-)
+# The above comment is from #20083
+./foo
+#dir1//foo
```

Comment:
why the double / ?
Let's keep the test simple and just make it `#dir1/foo` to make sure its really ignored.
And if you do end up supporting spaces before "#" then please include that in the test too.
And you should probably add one where "#" is in the middle of the line - to make sure we *don't* read it as a comment.